src/tests/src/study/system-model/test_system.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/study/system-model/system.h"
+
+#include "utils.h"
+
+using namespace Antares::Study::SystemModel;
+
+struct SystemTestFixture
+{
+    static Component createComponent(std::string id);
+};
```

Comment:
again, this is to avoid name clash of this function if we create another one in another test file with the same name. putting it into a unique fixture makes it unique

Reply:
**createComponent** has nothing to do with  **set up** or **tear down**, so it does not belong to a fixture.
As you know, tests have 3 steps : **Arrange**, **Act** and **Assert**. 
Calls to your **createComponent** belongs to the **Act** step.

We should make it a free function.
If you're afraid of link mismatch, declare it as **static**, that's what the keyword is for.

Besides, link mismatch are allowed because we have several unit tests pages in an executable. Is it suitable ?
But this is another subject.

Reply:
ok, i'll make it static meanwhile